--- a/src/examples/Makefile.am
+++ b/src/examples/Makefile.am
@@ -29,7 +29,7 @@
 @ELEMENTARY_X_LIBS@ \
 $(top_builddir)/src/lib/libelementary.la
 
-ELEMENTARY_CODEGEN = $(top_builddir)/src/bin/@ELEMENTARY_CODEGEN_PRG@
+ELEMENTARY_CODEGEN = @elementary_codegen@
 ELEMENTARY_CODEGEN_FLAGS = --prefix=codegen_example
 
 SRCS = \
@@ -154,10 +154,6 @@
 
 if EFL_BUILD_EXAMPLES
 files_DATA += \
-	prefs_example_01.epb \
-	prefs_example_02.epb \
-	prefs_example_03.epb \
-	prefs_example_03.edj \
 	theme_example.edj \
 	layout_example.edj\
 	codegen_example.edj
